server/data/data-library/src/main/java/org/apache/james/user/lib/AbstractUsersRepository.java

```diff
@@ -69,35 +70,43 @@ public void setDomainList(DomainList domainList) {
         this.domainList = domainList;
     }
 
-    protected void isValidUsername(String username) throws UsersRepositoryException {
-        Username user = Username.of(username);
+    protected void validateUsername(String username) throws UsersRepositoryException {
+        Username user = convert(username);
         if (supportVirtualHosting()) {
             // need a @ in the username
             if (!user.hasDomainPart()) {
-                throw new UsersRepositoryException("Given Username needs to contain a @domainpart");
+                throw new InvalidUsernameException("Given Username needs to contain a @domainpart");
             } else {
                 Domain domain = user.getDomainPart().get();
                 try {
                     if (!domainList.containsDomain(domain)) {
-                        throw new UsersRepositoryException("Domain does not exist in DomainList");
+                        throw new InvalidUsernameException("Domain does not exist in DomainList");
                     }
                 } catch (DomainListException e) {
-                    throw new UsersRepositoryException("Unable to query DomainList", e);
+                    throw new InvalidUsernameException("Unable to query DomainList", e);
                 }
             }
         } else {
             // @ only allowed when virtualhosting is supported
             if (user.hasDomainPart()) {
-                throw new UsersRepositoryException("Given Username contains a @domainpart but virtualhosting support is disabled");
+                throw new InvalidUsernameException("Given Username contains a @domainpart but virtualhosting support is disabled");
             }
         }
     }
 
+    private Username convert(String username) throws InvalidUsernameException {
+        try {
+            return Username.of(username);
+        } catch (IllegalArgumentException e) {
+            throw new InvalidUsernameException(e.getMessage(), e);
```

Comment:
Please don't wrap this runtime exceptions.

Ratioinals: With https://github.com/linagora/james-project/pull/2815 we will strongly type this API.

Hence webadlmin will need to supply a username then pass it to this class. This will make error handling outdated and generate rebase issues.

If you don't wrap this exception, things will go smoothly.